test/e2e_env/universal/transparentproxy/transparent_proxy.go

```diff
@@ -35,42 +35,35 @@ func TransparentProxy() {
 		Expect(universal.Cluster.DeleteMesh(mesh)).To(Succeed())
 	})
 
-	// This test is currently disabled as it basically never worked as expected
-	// as when installing tproxy again, it was just adding the same set of rules
-	// after the existing ones (with small exception to DNS ones, but it's
-	// irrelevant here). So the tests were passing only because rules form
-	// the second installation have been just ignored.
-	//
-	// When mechanism to uninstall tproxy will be implemented (ref.
-	// https://github.com/kumahq/kuma/issues/6093), we will adapt this test
-	// to run the uninstaller step in between.
-	XIt("should be able to re-install transparent proxy", func() {
+	It("should be able to re-install transparent proxy", func() {
 		// given
 		Eventually(func(g Gomega) {
 			_, err := client.CollectResponses(universal.Cluster, "tp-client", "test-server.mesh")
 			g.Expect(err).ToNot(HaveOccurred())
 		}).Should(Succeed())
 
 		// when
-
-		// This logic is currently non-existent
-		// TODO: remove above comment after implementing uninstaller logic
-		//       (ref. https://github.com/kumahq/kuma/issues/6093)
 		Eventually(func(g Gomega) {
 			stdout, _, err := universal.Cluster.Exec("", "", "tp-client",
-				"/usr/bin/kumactl", "uninstall", "transparent-proxy",
-				"--kuma-dp-user", "kuma-dp", "--verbose")
+				"/usr/bin/kumactl", "uninstall", "transparent-proxy", "--verbose")
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(stdout).To(ContainSubstring("transparent proxy cleanup completed successfully"))
+		}).Should(Succeed())
+
+		// then
+		Eventually(func(g Gomega) {
+			failures, err := client.CollectFailure(universal.Cluster, "tp-client", "test-server.mesh")
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stdout).To(ContainSubstring("Transparent proxy uninstalled successfully"))
+			g.Expect(failures.Exitcode).To((Or(Equal(6), Equal(28))))
```

Comment:
why are there 2 codes that can appear?

Reply:
maybe in the future we could have these codes mapped to something more readable / understandable

Reply:
Sometimes we can have timeout on a dns call so I prefer to keep both